#!/bin/sh /usr/share/dpatch/dpatch-run
## 05_libneon27_transition.dpatch by Kartik Mistry <kartik.mistry@gmail.com>
## DP: Patch for adding support for libneon27

@DPATCH@
--- sitecopy-0.16.3-orig/configure.in	2007-10-15 18:33:28.000000000 +0530
+++ sitecopy-0.16.3/configure.in	2007-10-15 18:35:05.000000000 +0530
@@ -112,8 +112,8 @@
    AC_LIBOBJ(lib/fnmatch)
 fi
 
-# Support neon 0.24, 0.25, 0.26.
-NE_REQUIRE_VERSIONS([0], [24 25 26])
+# Support neon 0.24, 0.25, 0.26, 0.27
+NE_REQUIRE_VERSIONS([0], [24 25 26 27])
 
 dnl But we don't use zlib or ACL support
 NEON_WITHOUT_ZLIB

--- sitecopy-0.16.3-orig/src/davdriver.c	2007-10-15 18:33:28.000000000 +0530
+++ sitecopy-0.16.3/src/davdriver.c	2007-10-16 15:47:30.000000000 +0530
@@ -43,13 +43,13 @@
 
 #include <ne_request.h>
 #include <ne_basic.h>
-#include <ne_basic.h>
 #include <ne_props.h>
 #include <ne_alloc.h>
 #include <ne_uri.h>
 #include <ne_auth.h>
 #include <ne_dates.h>
 #include <ne_socket.h>
+#include <ne_session.h>
 
 #include "protocol.h"
 #include "frontend.h"
@@ -140,13 +140,12 @@
 		       username, password);
 }
 
-static void notify_cb(void *userdata, ne_conn_status status, const char *info)
+static void notify_cb(void *userdata, ne_session_status status, const char *info)
 {
 
-#define MAP(a) case ne_conn_##a: fe_connection(fe_##a, info); break
+#define MAP(a) case ne_status_##a: fe_connection(fe_##a, info); break
 
     switch (status) {
-	MAP(namelookup);
 	MAP(connecting);
 	MAP(connected);
     default:
@@ -235,7 +234,7 @@
         }
     }
 
-    ne_set_status(sess, notify_cb, NULL);
+    ne_session_status(notify_cb);
 
     if (site->http_limit) {
 #if NE_VERSION_MINOR > 25
